doc/rst/developer/bestPractices/ContributorInfo.rst

```diff
@@ -9,80 +9,297 @@ The git repository for the project is hosted on Github at
+
+Request feedback on your changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. _Push your work:
+
+Push your work to your feature branch
++++++++++++++++++++++++++++++++++++++
+
+Push your changes to your feature branch on GitHub
+
+.. code-block:: bash
+
+    git push origin <branch_name>
+
+    # or if you don't like typing your complicated branch name,
+    # you can use this command to push the current branch:
+    git push origin HEAD
+
+    # if you forgot your branch name, you can get it by running
+    git branch
+
+    # it is the starred one...
+
+Note that if you have already created a pull request from a
+feature branch, pushing your work to that feature branch will
+update the pull request.
+
+This will enable others to see your work.
+
+.. _Ask for feedback early:
+
+Ask for feedback on your branch early (optional)
+++++++++++++++++++++++++++++++++++++++++++++++++
+
+Not ready to merge yours changes, but still want to see if your work is going
+in the right direction?  Feel free to ask for early feedback!
 
-* Develop your feature, bug fix, etc on your fork. Using a concisely named
-  branch is encouraged.
 
 .. _Submit pull request:
 
 Submit pull request
-~~~~~~~~~~~~~~~~~~~
++++++++++++++++++++
 
 * `Submit a pull request`_ with your changes (make sure you have `synced with
   the main repo`_).
 
-* In order for a pull request to be accepted and merged, it should a) pass
-  testing and b) be reviewed by a member of the core Chapel team (currently, a
-  member of the Cray Chapel team).
+  To do this, after pushing your changes to your feature branch on GitHub,
+  you can use the GitHub web interface to create a pull request. Visit
+
+  ``https://github.com/<username>/chapel``
+
+  and look for a "Compare & pull request" button for your feature branch.
+  Alternatively, navigate to your feature branch, and click the green icon next
+  to the branch dropdown to "Compare, review, create a pull request".
+
+  Next, put in a message to your reviewer about the purpose of your pull request
+  and give the pull request a useful title. It's a good time to draft the commit
+  message that you will need when merging the pull request.  See `Final merge
+  message`_ for recommendations on what that commit message should look like.
+
+  You will have to have signed a contributors agreement.
+
+  Your pull request will be available at a URL like:
+
+  ``https://github.com/chapel-lang/chapel/pull/<number>``
+
+  and you can discuss the patch with your reviewers there.
+
+  In working with your reviewers, you will no doubt change your pull request.
+  Just do your local development and then update your feature branch as in `Push
+  your work`_
+
+* In order for a pull request to be accepted and merged, it should
+
+  a) pass testing and
+  b) be reviewed by a member of the core Chapel team (currently, a member of the
+     Cray Chapel team).
 
 * It is considered good practice to keep PRs (pull requests) to a reasonable
```

Comment:
I think this section should move down to Policy and just be linked to there with a 1-liner like "It's a good idea to keep PRs reasonably sized".